packages/tools/src/stateManagement/annotation/AnnotationGroup.ts

```diff
@@ -0,0 +1,163 @@
+import { eventTarget, triggerEvent } from '@cornerstonejs/core';
+import Events from '../../enums/Events';
+import { getAnnotation } from './annotationState';
+
+export type BaseEventDetail = {
+  viewportId: string;
+  renderingEngineId: string;
+};
+
+/**
+ * An annotation group
+ */
+
+export default class AnnotationGroup {
+  private annotationUIDs = new Set<string>();
+  private _isVisible = true;
+  private _isActive = true;
+
+  public visibleFilter: (uid: string) => boolean;
+  protected addAnnotationListener: (evt) => void;
+
+  constructor() {
+    this.visibleFilter = this.unboundVisibleFilter.bind(this);
+    this.addAnnotationListener = this.unboundAddAnnotationListener.bind(this);
+  }
+
+  /**
+   * Returns true if other groups are free to hide this annotation.
+   * That is, if the annotation is not a member or is hidden.
+   */
+  protected unboundVisibleFilter(uid: string): boolean {
+    return !this._isVisible || !this.annotationUIDs.has(uid);
+  }
+
+  /**
+   * Event handler for adding new items to this group
+   * Does NOT add modified entries.
+   */
+  protected unboundAddAnnotationListener(evt) {
+    if (!this._isActive) {
+      return;
+    }
+    const { annotationUID } = evt.detail.annotation || evt.detail;
+    this.annotationUIDs.add(annotationUID);
+  }
+
+  public has(uid: string): boolean {
+    return this.annotationUIDs.has(uid);
+  }
+  /**
+   * Sets whether annotations belonging to this group are visible or not.
+   * If there are multiple groups, then the set visible false should be called
+   * before before re-enabling the other groups with setVisible true.
+   */
+  public setVisible(
+    isVisible = true,
+    baseEvent: BaseEventDetail,
+    filter?: (annotationUID: string) => boolean
+  ) {
+    if (this._isVisible === isVisible) {
+      return;
+    }
+    this._isVisible = isVisible;
+    this.annotationUIDs.forEach((uid) => {
+      const annotation = getAnnotation(uid);
+      if (!annotation) {
+        this.annotationUIDs.delete(uid);
+        return;
+      }
+      if (annotation.isVisible === isVisible) {
+        return;
+      }
+      if (!isVisible && filter?.(uid) === false) {
+        return;
+      }
+      annotation.isVisible = isVisible;
+      const eventDetail = {
+        ...baseEvent,
+        annotation,
+      };
+      triggerEvent(eventTarget, Events.ANNOTATION_MODIFIED, eventDetail);
+    });
+  }
+
+  public get isVisible() {
+    return this._isVisible;
+  }
+
+  public get isActive() {
+    return this._isActive;
+  }
+
+  /** Finds the nearby/next annotation in the given direction */
+  public findNearby(uid: string, direction: 1) {
+    if (!this._isVisible) {
+      return null;
+    }
```

Comment:
I don't think there is any need for this here. One should be able to get nearby annotations even if they are invisible